src/main/kotlin/org/rust/cargo/project/model/impl/CargoSyncTask.kt

```diff
@@ -315,15 +313,16 @@ private fun fetchCargoWorkspace(context: CargoSyncTask.SyncContext, rustcInfo: R
         val cargo = toolchain.cargoOrWrapper(projectDirectory)
         try {
             CargoEventService.getInstance(childContext.project).onMetadataCall(projectDirectory)
-            val projectDescriptionData = cargo.fullProjectDescription(childContext.project, projectDirectory, object : ProcessAdapter() {
-                override fun onTextAvailable(event: ProcessEvent, outputType: Key<Any>) {
-                    val text = event.text.trim { it <= ' ' }
-                    if (text.startsWith("Updating") || text.startsWith("Downloading")) {
-                        childContext.withProgressText(text)
-                    }
-
+            val projectDescriptionData = cargo.fullProjectDescription(
+                childContext.project,
+                projectDirectory,
+            ) {
+                when (it) {
+                    CargoCallType.METADATA -> SyncProcessAdapter(childContext)
+                    // TODO: collect build events and show them in structured way
+                    CargoCallType.BUILD_SCRIPT_CHECK -> SyncProcessAdapter(childContext)
```

Comment:
They were added to support build events during build script evaluation in my initial branch (although I'm not sure that we need lambda there). Since I extracted only part about metadata fetching, I've dropped this change at all for now